tools/integration_tests/util/client/storage_client.go

```diff
@@ -97,3 +102,57 @@ func CreateObjectOnGCS(ctx context.Context, client *storage.Client, object, cont
 
 	return nil
 }
+
+// Create storage client with a configurable timeout and retrun a function to cancel the storage client
+func CreateStorageClientWithTimeOut(ctx *context.Context, storageClient **storage.Client, time time.Duration) (func() error, error) {
+	var err error
+	var cancel context.CancelFunc
+	*ctx, cancel = context.WithTimeout(*ctx, time)
+	*storageClient, err = CreateStorageClient(*ctx)
+	if err != nil {
+		log.Fatalf("client.CreateStorageClient: %v", err)
+	}
+
+	// return func to close storage client and release resources.
+	return func() error {
+		err := (*storageClient).Close()
+		defer cancel()
+		if err != nil {
+			return fmt.Errorf("Failed to close storage client: %v", err)
+		}
+		return nil // No error during close
+	}, nil // No error during setup
+}
+
+// Downloads an object to a file.
+func DownloadObjectFromGCS(gcsFile string, destFileName string, t *testing.T) error {
```

Comment:
We can remove the dependency of t *testing.T from here also.
It's okay, to directly call open here.

Later, we can use the cache/util.CreateFile here.

Reply:
As talked offline we will keep with `t *testing.T` as we are using this function in functional test.